src/query/expression/src/function.rs

```diff
@@ -651,9 +653,18 @@ where F: Fn(&[ValueRef<AnyType>], &mut EvalContext) -> Value<AnyType> {
             Value::Column(column) => {
                 let result = match column {
                     Column::Nullable(box nullable_column) => {
-                        let validity = bitmap.into();
-                        let validity =
-                            common_arrow::arrow::bitmap::and(&nullable_column.validity, &validity);
+                        let mut before_map = bitmap.clone().into();
+                        let mut after_map = nullable_column.validity.clone();
+                        match bitmap.len().cmp(&after_map.len()) {
+                            Ordering::Greater => {
+                                after_map = expand_bitmap(&after_map, bitmap.len());
```

Comment:
Maybe we can just use `bitmap.sliced(offset, length)` to get better performance.

Reply:
yes, sliced is a good idea.